packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,281 @@
+      1,
+    );
+  }, [taskState]);
+
+  React.useEffect(() => {
+    if (!taskState || !taskState.phases || !taskState.active) {
+      setNavigationStart(null);
+      setNavigationDestination(null);
+      return;
+    }
+
+    const message = Object.values(taskState.phases)[taskState.active - 1]?.detail;
+
+    if (message) {
+      const regex = /\[place:(.*?)\]/g;
+
+      let match;
+      const waypoints = [];
+
+      // Iterate over all matches found by the regular expression
+      while ((match = regex.exec(message.toString()))) {
+        waypoints.push(match[1]);
+      }
+
+      setNavigationStart(waypoints[0]);
+      setNavigationDestination(waypoints[1]);
+    } else {
+      setNavigationStart('-');
+      setNavigationDestination('-');
+      console.error("Failed to retrieve robot's current navigation start and destination.");
+    }
+  }, [taskState]);
+
+  const returnDialogContent = () => {
+    const contents = [
+      {
+        title: 'Assigned tasks',
+        value: taskState ? taskState.booking.id : 'No task',
+      },
+      {
+        title: 'Est. end time',
+        value: taskState?.unix_millis_finish_time
+          ? `${new Date(taskState?.unix_millis_finish_time).toLocaleString()}`
+          : '-',
+      },
+    ];
+
+    if (taskState) {
+      contents.push(
+        {
+          title: 'Navigation start',
+          value: navigationStart ? navigationStart : '-',
+        },
+        {
+          title: 'Navigation destination',
+          value: navigationDestination ? navigationDestination : '-',
+        },
+      );
+    }
+
+    return (
+      <>
+        {contents.map((message, index) => (
+          <div key={index}>
+            <TextField
+              label={message.title}
+              id="standard-size-small"
+              size="small"
+              variant="filled"
+              InputProps={{ readOnly: true, className: classes.textField }}
+              fullWidth={true}
+              multiline
+              maxRows={4}
+              margin="dense"
+              value={message.value}
+            />
+          </div>
+        ))}
+      </>
+    );
+  };
+
+  return (
+    <Dialog
+      PaperProps={{
+        style: {
+          backgroundColor: setTaskDialogColor(robotState?.status),
+          boxShadow: 'none',
+        },
+      }}
+      open={isOpen}
+      onClose={() => {
+        setIsOpen(false);
+        onClose();
+      }}
+      fullWidth
+      maxWidth="sm"
+    >
+      <Grid container mb={1} alignItems="center" spacing={1}>
+        <Grid item xs={2}></Grid>
+        <Grid item xs={8}>
+          <DialogTitle align="center">Robot summary: {robotState?.name}</DialogTitle>
+        </Grid>
+        <Grid item xs={2}>
+          <Grid container justifyContent="flex-end">
+            <Typography variant="subtitle1">{`${
+              robotState?.battery ? robotState?.battery * 100 : 0
```

Comment:
We will need to clip decimal points, 
![image](https://user-images.githubusercontent.com/5383623/236135601-26f5c4d5-0db8-46ac-bcc5-752a1add9d4e.png)